tripal_chado/includes/loaders/tripal_chado.pub_importer_AGL.inc

```diff
@@ -553,6 +553,36 @@ function tripal_pub_AGL_count($yazc, $search_str) {
   return $count;
 }
 
+/**
+ * Decode the unusal text encoding returned from our
+ * call to yaz_record(..., 'xml; charset=marc-8,utf-8')
+ * Some characters are in UTF-8, some are encoded as HTML
+ * entities, and some HTML entities are double-encoded,
+ * for example &amp;#x2018;
+ * A straight call to mb_convert_encoding() will corrupt
+ * any UTF-8 characters, so only convert what appears
+ * to be an HTML entity
+ *
+ * @param $text
+ *   The string to be decoded to "pure" UTF-8
+ *
+ * @return
+ *   The decoded string
+ *
+ * @ingroup tripal_pub
+ */
+function tripal_pub_AGL_decode($text) {
+  // first handle double encoding situations by replacing &amp;
+  $text = preg_replace("/&amp;/", "&", $text);
+
+  // then only replace things that look like an HTML entity, i.e.
+  // ampersand followed by semicolon, in order to leave UTF-8 intact
+  $text = preg_replace_callback("/(&[^;\p{C}\p{M}\p{Z}]{1,31};)/",
```

Comment:
if the goal is to decode html entities, [can we use `html_entity_decode` instead?](https://www.php.net/manual/en/function.html-entity-decode.php)

Reply:
Thank you Bradford, I didn't know about that function, it is a much cleaner way to do it. I have updated and tested with this new method.